Add cToken strategy

arr00 opened the pull request:
Adds the cToken strategy. See  src/strategies/cToken/README.md for details.

bonustrack commented on src/strategies/cToken/index.ts:
```diff
@@ -0,0 +1,94 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+
+export const author = 'Arr00';
```
The author field should be your Github username with the same casing.
```suggestion
export const author = 'arr00';
```

bonustrack commented on src/strategies/cToken/index.ts:
```diff
@@ -0,0 +1,94 @@
+    stateMutability: 'view',
+    type: 'function'
+  }
+
```
```suggestion
```

bonustrack commented on src/strategies/cToken/index.ts:
```diff
@@ -0,0 +1,94 @@
+export async function strategy(network, provider, addresses, options, snapshot) {
+  let blockTag = typeof snapshot === 'number' ? (snapshot) : 'latest';
+ 
+
```
```suggestion
```

bonustrack commented on src/strategies/index.ts:
```diff
@@ -27,5 +28,6 @@ export default {
   pancake,
-  synthetix
+  synthetix,
+  cToken
```
The name of the strategy should be lowercase, you can use either `ctoken` or `c-token`

bonustrack commented on src/strategies/cToken/index.ts:
```diff
@@ -0,0 +1,94 @@
+  let blockTag = typeof snapshot === 'number' ? (snapshot) : 'latest';
+ 
+
+  const [balanceNowResponse, borrowResponse]  = await Promise.all([
```
It's better if we can reduce the number of request being sent. I recommend to merge the 2 multicall requests into a single multicall request.

bonustrack commented on src/strategies/cToken/index.ts:
```diff
@@ -0,0 +1,94 @@
+    if(options.borrowingRestricted) {
+      noBorrow = borrowResponse[i].toString().localeCompare('0') == 0 ? 1:0;
+    }
+    resultData[addresses[i]] = Math.min(parseFloat(formatUnits(balanceNowResponse[i].toString(), options.decimals)), parseFloat(formatUnits(balanceOldResponse[i].toString(), options.decimals))) * noBorrow;
```
This line is long and hard to read, would be better if you can use vars like this
```suggestion
    resultData[addresses[i]] = Math.min(balanceNow, balanceOld) * noBorrow;
```

bonustrack reviewed (commented):
@arr00 I've added few suggestions. Can you provide an address that will return a positive score so we can test the strategy? Also you may want to run the lint script with "npm run lint" Thank you.

arr00 commented:
I addressed all the comments I believe. Please take a look again. Any cUNI holder that isn't borrowing UNI should return a non-zero score. Here is an example of one that you can use for testing: 0x161fac24d54698755dab0fcd65e2c883928ca724 as of block 11147224.

bonustrack reviewed (approved):
Looking great!
